src/DocumentFormat.OpenXml/SimpleTypes/OpenXmlComparableSimpleReference.cs

```diff
@@ -0,0 +1,172 @@
+        }
+
+        /// <summary>
+        /// Gets or sets the value of this instance.
+        /// </summary>
+        public abstract T Value { get; set; }
+
+        /// <inheritdoc />
+        public int CompareTo(object obj)
+        {
+            switch (obj)
+            {
+                case null:
+                    return 1;
+
+                case OpenXmlComparableSimpleReference<T> other:
+                    return CompareTo(other);
+            }
+
+            string message = string.Format(ExceptionMessages.IncompatibleArgumentType, obj.GetType().FullName);
+            throw new ArgumentException(message);
+        }
+
+        /// <inheritdoc />
+        public int CompareTo(OpenXmlComparableSimpleReference<T> other)
+        {
+            if (other is null)
+            {
+                return 1;
+            }
+
+            if (Value == null)
+            {
+                return other.Value == null ? 0 : -1;
+            }
+
+            return Value.CompareTo(other.Value);
+        }
+
+        /// <inheritdoc />
+        public bool Equals(OpenXmlComparableSimpleReference<T> other)
+        {
+            return !(other is null) && Equals(Value, other.Value);
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            switch (obj)
+            {
+                case null:
+                    return false;
+
+                case OpenXmlComparableSimpleReference<T> other:
+                    return Equals(other);
+
+                case T otherValue:
+                    return otherValue.Equals(Value);
+
+                default:
+                    return false;
+            }
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            return Value?.GetHashCode() ?? 0;
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="left"></param>
+        /// <param name="right"></param>
+        /// <returns></returns>
+        public static bool operator ==(OpenXmlComparableSimpleReference<T> left, OpenXmlComparableSimpleReference<T> right)
+        {
+            if (left is null)
+            {
+                return right is null;
+            }
+
+            return left.Equals(right);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="left"></param>
+        /// <param name="right"></param>
+        /// <returns></returns>
+        public static bool operator !=(OpenXmlComparableSimpleReference<T> left, OpenXmlComparableSimpleReference<T> right)
+        {
+            return !(left == right);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="left"></param>
+        /// <param name="right"></param>
+        /// <returns></returns>
+        public static bool operator <(OpenXmlComparableSimpleReference<T> left, OpenXmlComparableSimpleReference<T> right)
+        {
```

Comment:
What about the comments on those operators (==, !=, <, <=, >, >=)? I have not completed them but don't know whether we really need them.

Reply:
hmmm for completeness, either fill them in (probably some boiler plate comments), or just remove them (unless if stylecop complains - I can't remember if I've turned those on yet)

Reply:
I added those empty comments because Style Cop did complain. I will steal something elsewhere.